cinch/roles/jenkins_master/templates/etc/nginx/conf.d/jenkins_http.conf

```diff
@@ -0,0 +1,46 @@
+server_names_hash_bucket_size 6400;
+upstream jenkins {
+  server 127.0.0.1:8080 fail_timeout=0;
+}
+
+server {
+  listen 80; 
+  server_name {{ inventory_hostname }}; 
```

Comment:
Well some examples of a place we don't use either DNS hostname or IP address would be:

* Our Vagrant files use a hostname equal to the Vagrantfile name and set ansible_host
* Our Docker images that get an inventory_hostname equal to their container name
* Machines generated by linchpin which get their topology names configured as inventory hostname and an appropriate value configured to ansible_host
* Anyone who wants to shortcut group names by setting the inventory_hostname to something like "jenkins_master" and then set ansible_host to an fqdn or IP address
* Someone who is configuring behind a firewall or load balancer where hosts have internal-only names and not names that can be reached from the Ineternet
* Basically every single Ansible inventory I've ever worked with anywhere.

You should factor this out to a single value that a user can set but which draws a default value from some set of either the Ansible facts (you can discover the default IP address for the system - see the vagrant directory for some examples of where I do this) or from the inventory but default to the value of ansible_host if it is set (see the check_ssh role for an example of how to do that). But just blindly using inventory_hostname is going to break things.